src/v/iceberg/values_avro.cc

```diff
@@ -113,14 +128,28 @@ struct primitive_value_avro_visitor {
         }
         return {data};
     }
-    avro::GenericDatum operator()(const decimal_value&) {
-        throw std::invalid_argument("XXX decimal not implemented");
+    avro::GenericDatum operator()(const decimal_value& v) {
+        std::vector<uint8_t> bytes(16);
+        auto low = absl::Int128Low64(v.val);
+        auto high = absl::Int128High64(v.val);
+
+        // Store in little-endian order (least significant byte first)
```

Comment:
The avro spec mentions this should be big-endian, assuming they're referring to the fixed type representation too: https://avro.apache.org/docs/1.12.0/specification/

>A decimal logical type annotates Avro bytes or fixed types. The byte array must contain the two’s-complement representation of the unscaled integer value in big-endian byte order. The scale is fixed, and is specified using an attribute.

Looked at the Java impl and it looks to be the case as well:
- https://github.com/apache/avro/blob/80400781a796bc0e90dd8ea1db42234926db33e9/lang/java/avro/src/main/java/org/apache/avro/Conversions.java#L128-L141
- https://docs.oracle.com/javase/8/docs/api/?java/math/BigInteger.html

Reply:
done